woocommerce/wc-admin
# Add `reports/downloads` REST API endpoint.

See #428. This implements one of the required endpoints needed for the downloads report. It implements https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_downloads, in addition to the required filter options that were missing from the schema.

### Detailed test instructions:

- Run `phpunit` and make sure all tests are passing.

Extra steps:
- Create two different test downloadable products. Make sure a file is attached. https://docs.woocommerce.com/document/digital-downloadable-product-handling/
- Place some orders, and purchase access to the download.
- Under your stores front end, go to "My Account", and download each file at least once.
- Use Postman or another tool to make some test requests to `/wc/v3/reports/downloads`. You can use various filters with your product, order, or user IDs to test the filters.
- Spot check the responses, spot check the headers for pagination info.
